pkg/apis/awsprovider/v1alpha1/awsclusterproviderconfig_types.go

```diff
@@ -57,6 +58,10 @@ type AWSClusterProviderSpec struct {
 	// ClusterConfiguration holds the cluster-wide information used during a
 	// kubeadm init call.
 	ClusterConfiguration kubeadmv1beta1.ClusterConfiguration `json:"clusterConfiguration,omitempty"`
+
+	// AdditionalUserDataFiles specifies extra files to be passed to Machines upon creation.
+	// +optional
+	AdditionalUserDataFiles []userdata.Files `json:"additionalUserDataFiles,omitempty"`
```

Comment:
I know this wasn't specced out anywhere, but I wonder if we want to do this at the machine level vs cluster (or both)? I would imagine that we wouldn't need (or even want) e.g. a client certificate used to authenticate to a webhook service to be placed on all machines.

Reply:
maybe both is the way to go

Reply:
Added and updated the PR